language_generators/scala-defref-extractor/src/test/scala/io/bazeltools/buildgen/scaladefref/CanParseFileTest.scala

```diff
@@ -17,41 +17,62 @@ class CanParseFileTest extends AnyFunSuite {
     case class Cat(foo: String)
     """
     val expectedDataBlock = DataBlock(
-         "",
-        defs = SortedSet(Entity.dotted("com.foo.bar.Cat"), Entity.dotted("com.foo.bar.Cat.foo")),
-        refs = SortedSet(Entity.dotted("String"), Entity.dotted("com.foo.bar.String")),
-        bzl_gen_build_commands = SortedSet()
+      "",
+      defs = SortedSet(
+        Entity.dotted("com.foo.bar.Cat"),
+        Entity.dotted("com.foo.bar.Cat.foo")
+      ),
+      refs =
+        SortedSet(Entity.dotted("String"), Entity.dotted("com.foo.bar.String")),
+      bzl_gen_build_commands = SortedSet()
     )
     assertParse(simpleContent, expectedDataBlock)
   }
 
-    test("can extract a strange syntax") {
+  test("can extract a strange syntax") {
     val simpleContent = """
     package com.foo.bar
+    import com.example.Wolf
+    import com.example.Elephant
 
-    case class Cat(foo: String) {
+    case class Cat(foo: String) extends Wolf with Elephant {
         val expressionName = Dog()
         // This format means we will throw a match error if Lizard != Dog with match
         val (`expressionName`, pie) = (Lizard(), 33)
     }
     """
 
-
-
-    val expectedDataBlock = DataBlock("", SortedSet(
-            Entity.dotted("com.foo.bar.Cat"),
-            Entity.dotted("com.foo.bar.Cat.expressionName"),
-            Entity.dotted("com.foo.bar.Cat.foo"),
-            Entity.dotted("com.foo.bar.Cat.pie")
-            ),
-            SortedSet(
-                Entity.dotted("Dog"),
-                Entity.dotted("Lizard"),
-                Entity.dotted("String"),
-                Entity.dotted("com.foo.bar.Dog"),
-                Entity.dotted("com.foo.bar.Lizard"),
-                Entity.dotted("com.foo.bar.String")
-                ), SortedSet())
+    val expectedDataBlock = DataBlock(
+      "",
+      SortedSet(
+        Entity.dotted("com.foo.bar.Cat"),
+        Entity.dotted("com.foo.bar.Cat.expressionName"),
+        Entity.dotted("com.foo.bar.Cat.foo"),
+        Entity.dotted("com.foo.bar.Cat.pie")
+      ),
+      SortedSet(
+        Entity.dotted("Dog"),
+        Entity.dotted("Lizard"),
+        Entity.dotted("String"),
+        Entity.dotted("com.foo.bar.Dog"),
+        Entity.dotted("com.foo.bar.Lizard"),
+        Entity.dotted("com.foo.bar.String"),
+        Entity.dotted("com"),
+        Entity.dotted("com.example"),
+        Entity.dotted("com.example.Elephant"),
+        Entity.dotted("com.example.Wolf"),
+        Entity.dotted("com.foo.bar.com"),
+        Entity.dotted("com.foo.bar.com.example"),
+        Entity.dotted("com.foo.bar.com.example.Elephant"),
+        Entity.dotted("com.foo.bar.com.example.Wolf")
+      ),
+      SortedSet(
```

Comment:
I'm not following how this is populated now since DataBlock ignores addLink.

Oh... is this test actually run in CI? I don't think so.